cmd/kapacitord/main.go

```diff
@@ -87,16 +88,38 @@ func (m *Main) Run(args ...string) error {
 		}
 
 		signalCh := make(chan os.Signal, 1)
-		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
 		m.Diag.Info("listening for signals")
 
-		// Block until one of the signals above is received
-		select {
-		case <-signalCh:
-			m.Diag.Info("signal received, initializing clean shutdown...")
-			go func() {
-				cmd.Close()
-			}()
+	Loop:
+		for s := range signalCh {
+			switch s.String() {
+			case syscall.SIGTERM.String():
+				m.Diag.Info("SIGTERM received, initializing clean shutdown...")
+				go func() {
+					cmd.Close()
+				}()
+				break Loop
+
+			case syscall.SIGHUP.String():
+				m.Diag.Info("SIGHUP received, reloading tasks/templates/handlers directory...")
+				if err := cmd.Server.LoadService.Load(); err != nil {
```

Comment:
I was thinking, instead of doing cmd.Server.LoadService.Load() and handing the error types here in the main package, create a method on the server called Reload or something, that if it returns an error at all causes the Server to crash.
Then inside that method determine if the underlying errors returned from LoadService.Load are work crashing. That way the main function crashes if it gets any error, and the Server contains the logic on what errors cause crashes.

Does that help?

Reply:
yup.